atomic_reactor/util.py

```diff
@@ -661,18 +667,23 @@ def default(self):
         """
         return self.v2_list or self.oci or self.v2 or self.v1
 
+    def __getattr__(self, attr):
+        return self.get(attr, None)
+
 
 def get_manifest_media_type(version):
-    if version in ('v1', 'v2'):
-        return 'application/vnd.docker.distribution.manifest.{}+json'.format(version)
-    elif version == 'v2_list':
-        return 'application/vnd.docker.distribution.manifest.list.v2+json'
-    elif version == 'oci':
-        return 'application/vnd.oci.image.manifest.v1+json'
+    if version in ManifestDigest.content_type.keys():
+        return ManifestDigest.content_type[version]
     else:
         raise RuntimeError("Unknown manifest schema type")
 
 
+def get_manifest_media_version(digest):
+    for version in ManifestDigest.content_type.keys():
+        if getattr(digest, version) == digest.default:
+            return version
```

Comment:
Should it raise an exception if it wasn't found?

Reply:
I'm not sure if this situation is realistic.  If no version has been set, `default` would return `None` - and it would match `v1` as its the first tag.

The only case where exception can be thrown is `def default` doesn't list all items from `content_type` - in this case tests would be broken.